### ISSUE TITLE
Playwright : If fullyParallel is set to false in config.ts file  {fullyParallel: false} the storage state is used only for the first test execution and next scenarios are skipped due to authentication.

### ISSUE THREAD

**Issue by SKannoth** [microsoft/playwright#28287]
When I force the test to execute in series and set the flag fullyParallel: false, in playwright.config.ts file and run the test the storage state file is used only for 1 single scenarios and the rest of the scenarios are skipped/failed due to authentication. 

### Source code

```js

import { defineConfig, devices } from '@playwright/test';

/**
 * Read environment variables from file.
 * https://github.com/motdotla/dotenv
 */
// require('dotenv').config();

/**
 * See https://playwright.dev/docs/test-configuration.
 */
require('dotenv').config();
export default defineConfig({
  globalSetup: "tests/utility/gloabl-setup.ts",
  testDir: './tests/specs',
  
  /* Run tests in files in parallel */
  fullyParallel: false,
  /* Fail the build on CI if you accidentally left test.only in the source code. */
  forbidOnly: !!process.env.CI,
  /* Retry on CI only */
  retries: process.env.CI ? 2 : 0,
  /* Opt out of parallel tests on CI. */
  workers: process.env.CI ? 1 : undefined,
  /* Reporter to use. See https://playwright.dev/docs/test-reporters */
  // reporter: 'html',
  /* Shared settings for all the projects below. See https://playwright.dev/docs/api/class-testoptions. */
  use: {
      baseURL: 'https://trading.dev.habitatenergy.online/',
      browserName: "chromium",
      storageState: "loginAuth.json",
      headless: false,

      extraHTTPHeaders: {
       
      'Accept': 'application/vnd.github.v3+json',
      
    },
    /* Collect trace when retrying the failed test. See https://playwright.dev/docs/trace-viewer */
    trace: 'on-first-retry',
  },

  /* Configure projects for major browsers */
  projects: [
    {
      name: 'chromium',
      use: { ...devices['Desktop Chrome'] },
    },

    // {
    //   name: 'firefox',
    //   use: { ...devices['Desktop Firefox'] },
    // },

    // {
    //   name: 'webkit',
    //   use: { ...devices['Desktop Safari'] },
    // },

    /* Test against mobile viewports. */
    // {
    //   name: 'Mobile Chrome',
    //   use: { ...devices['Pixel 5'] },
    // },
    // {
    //   name: 'Mobile Safari',
    //   use: { ...devices['iPhone 12'] },
    // },

    /* Test against branded browsers. */
    // {
    //   name: 'Microsoft Edge',
    //   use: { ...devices['Desktop Edge'], channel: 'msedge' },
    // },
    // {
    //   name: 'Google Chrome',
    //   use: { ...devices['Desktop Chrome'], channel: 'chrome' },
    // },
  ],

  /* Run your local dev server before starting the tests */
  // webServer: {
  //   command: 'npm run start',
  //   url: 'http://127.0.0.1:3000',
  //   reuseExistingServer: !process.env.CI,
  // },
});
```

**Comment by yury-s** [comment]
Can you please share reduced repro where the error happens and detailed steps to reproduce it. It's hard to understand from what your description what you are doing and what is not working as expected.

**Comment by SKannoth** [comment]
@yury-s please find the below sample code 

I have spec file which has couple of scenarios say:

```
test("Validate the Dispatch Page and verify the Assets", async({ page })=>{
    console.log('The seconf test cases')
    const dispatch = new Dispatch(page)
    const login = new Login(page)
    await login.navigate('/**/**')
    await page.waitForTimeout(3000)
    expect(await dispatch.peakersMenu().isVisible()).toBeTruthy()  
    const sadanam =  await dispatch.assetsList()
    await page.waitForTimeout(1000) 
    await page.getByText('check_box_outline_blank').click()
    await page.waitForTimeout(10000)
    const afterSearch = (await dispatch.assetsList().allTextContents()).toString().trim()
    expect(afterSearch).toBe('Marty')
})

test("001-Validate the DFR Outages Page and report outage", async({ page })=>{
    console.log('The seconf test cases')
    const login = new Login(page)
    const outages = new Outages(page)
    await login.navigate('/operations/outages')
    await page.waitForTimeout(500)
    expect(await outages.endDate().isVisible()).toBeTruthy()
    expect(await outages.startDate().isVisible()).toBeTruthy()
    await page.waitForTimeout(5000)
    await outages.reportOutage().click()
    await page.waitForTimeout(2000)
    await outages.batteriesaffected().click()
    await outages.affectedBatteryMarty().click()
    await outages.submitButton().click()

})

test("002-Verify the existing Reported and Delete it", async({ page })=>{
    const login = new Login(page)
    const outages = new Outages(page)
    await login.navigate('/operations/outages')
    await page.waitForTimeout(5000)
})
```

I have respective page elements in the page class

import { Page } from "playwright"


export default class Dispatch{

    page: Page

    constructor(page: Page){
        this.page = page
    }  

    assetsList(){
        return this.page.locator("//*[@class='title-style']")
    }

    searchbarMenu(){
        return this.page.getByText('***')
    }

} 

I have global config were I am using the authentication and using the storage state:

```
import { Browser, Page, chromium } from "playwright";
import { FullConfig } from "playwright/test";
import Login from "../pages/login.page";


async function gloablSetup(_config: FullConfig) {
    const browser:Browser = await chromium.launch({headless: false})
    const context = await browser.newContext()
    const page: Page = await context.newPage()
    const login = new Login(page)
    await login.navigate('**')
    await login.enterEmail(process.env.USERNAME!)
    await login.continueClick()
    await login.enterPassword(process.env.PASSWORD!)
    await login.loginClick()
    await page.context().storageState({ path: "loginAuth.json"})
    await page.waitForTimeout(3000)
    await browser.close()
    
}
export default gloablSetup;
```

When I run the script with "  fullyParallel: true," in config file all the scripts are executed and are using the "loginAuth.json" cookie, but If I want to execute it sequentially and set the flag  fullyParallel: false then only 1 scenario is getting executed and next 2 is failed due to authentication

**Comment by pavelfeldman** [comment]
Closing as per above, please feel free to open a new issue if this does not cover your use case.
